ObjCTraining/AddPropertyInfoViewController.m

```diff
@@ -0,0 +1,331 @@
+//
+//  AddPropertyInfoViewController.m
+//  ObjCTraining
+//
+//  Created by RWuser on 25/03/16.
+//  Copyright © 2016 RWuser. All rights reserved.
+//
+#import <QuartzCore/QuartzCore.h>
+#import "AddPropertyInfoViewController.h"
+#import "ValidationManager.h"
+
+typedef NS_ENUM( NSInteger, InputFieldsEnum) {
+    Address1FieldTag = 1,
+    Address2FieldTag,
+    CityFieldTag,
+    StateFieldTag,
+    ZipFieldTag,
+    CountryFieldTag
+};
+
+@interface AddPropertyInfoViewController()
+@property UIView *activeField;
+@end
+
+@implementation AddPropertyInfoViewController
+BOOL isFieldEmpty = NO;
+NSString * emptyFieldErrorMessage = @"Please Enter Full Address";
+NSString * wrongFieldErrorMessage = @"Please Enter Correct Address";
+
+NSMutableDictionary *isFieldsEmpty;
+NSMutableDictionary *isFieldsWrong;
+
+- (void)viewWillAppear:(BOOL)animated {
+    self.address1TextField.delegate = self;
+    self.address2TextField.delegate = self;
+    self.cityTextField.delegate = self;
+    self.stateTextField.delegate = self;
+    self.zipTextField.delegate = self;
+    self.countryTextField.delegate = self;
+    
+    [self registerForKeyboardNotifications];
+    [self.address1TextField becomeFirstResponder];
+    
+    isFieldsEmpty = [[NSMutableDictionary alloc] init];
+    isFieldsEmpty[[NSNumber numberWithInt:Address1FieldTag]] = @YES;
+    isFieldsEmpty[[NSNumber numberWithInt:Address2FieldTag]] = @YES;
+    isFieldsEmpty[[NSNumber numberWithInt:CityFieldTag]] = @YES;
+    isFieldsEmpty[[NSNumber numberWithInt:StateFieldTag]] = @YES;
+    isFieldsEmpty[[NSNumber numberWithInt:ZipFieldTag]] = @YES;
+    isFieldsEmpty[[NSNumber numberWithInt:CountryFieldTag]] = @YES;
+
+    isFieldsWrong = [[NSMutableDictionary alloc] init];
+    isFieldsWrong[[NSNumber numberWithInt:Address1FieldTag]] = @NO;
+    isFieldsWrong[[NSNumber numberWithInt:Address2FieldTag]] = @NO;
+    isFieldsWrong[[NSNumber numberWithInt:CityFieldTag]] = @NO;
+    isFieldsWrong[[NSNumber numberWithInt:StateFieldTag]] = @NO;
+    isFieldsWrong[[NSNumber numberWithInt:ZipFieldTag]] = @NO;
+    isFieldsWrong[[NSNumber numberWithInt:CountryFieldTag]] = @NO;
+    
```

Comment:
Also, it's a bad idea to fill your containers in viewWillAppear. Read about VC lifecycle.